src/OpenCSGRenderer.cc

```diff
@@ -46,153 +39,470 @@ class OpenCSGPrim : public OpenCSG::Primitive
 	EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 	OpenCSGPrim(OpenCSG::Operation operation, unsigned int convexity, const OpenCSGRenderer &renderer) :
 			OpenCSG::Primitive(operation, convexity), csgmode(Renderer::CSGMODE_NONE), renderer(renderer) { }
-	shared_ptr<const Geometry> geom;
+	shared_ptr<const PolySet> geom;
```

Comment:
I believe this was `boost::shared_ptr` originally, but can be updated to `std::shared_ptr`. This is baseline code.

Reply:
`src/memory.h` (which happens to be included via csgnode.h in this case) has this line and various other `using`s for convenience:
https://github.com/openscad/openscad/blob/4cfbf9985ed25b8ed919d0e59d5af31acdcbe21a/src/memory.h#L4

It would probably make sense to include `memory.h` directly from OpenCSGRenderer.cc for clarity.